app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,191 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@DisplayName("TaskService 클래스")
+class TaskServiceTest {
+    abstract class Context_didCreateTwoTasks {
+        final int numberOfTasksCreated = 2;
+
+        @BeforeEach
+        void context() {
+            for (int i = 1; i <= numberOfTasksCreated; i++) {
+                final Task task = new Task(FIXTURE_TITLE + i);
+                service.createTask(task);
+            }
+        }
+    }
+
+    public static final String FIXTURE_TITLE = "title";
+    private TaskService service;
+
+    @BeforeEach
+    void setup() {
+        service = new TaskService();
+    }
+
+    @Nested
+    @DisplayName("getTaskList 메소드는")
+    class Describe_getTaskList {
+        @Nested
+        @DisplayName("생성되어 있는 할 일이 없다면")
+        class Context_didNotCreateTask {
+            @BeforeEach
+            void prepare() {
+                service.createTask(new Task(FIXTURE_TITLE));
+                service.deleteTask(1L);
+            }
```

Comment:
이렇게 할 수 있다면 테스트 설명과 좀 더 자연스럽게 어울리겠죠.

```suggestion
        @Nested
        @DisplayName("생성되어 있는 할 일이 없다면")
        class Context_didNotCreateTask {
            @BeforeEach
            void prepare() {
                service.deleteAllTasks();
            }
```

`deleteAllTasks` 메소드를 만들 필요가 없다면 테스트 코드 스코프에서만 통하는 삭제 전용 컴포넌트를 만들어 쓰는 방법도 생각할 수 있습니다.

Reply:
[5e38c65](https://github.com/CodeSoom/spring-week3-assignment-1/pull/81/commits/5e38c65c1c132c280935f1026a3ff328f5d80b84) 이게 조언해주셨던 방향이 맞는지 모르겠지만 일단 시도해봤습니다.

Service에 바로 deleteAllTasks를 추가할 수 있겠지만, 실제 제품을 위한 메소드는 아니니까 deleteAllTasks를 Service에 추가하지 않는 방식으로 만들어봤습니다.